airbyte/cloud/_workspaces.py

```diff
@@ -0,0 +1,351 @@
+    from airbyte_api.models.shared.destinationresponse import DestinationResponse
+
+    from airbyte.caches.base import CacheBase
+
+
+@dataclass
+class CloudWorkspace:
+    """A remote workspace on the Airbyte Cloud.
+
+    By overriding `api_root`, you can use this class to interact with self-managed Airbyte
+    instances, both OSS and Enterprise.
+    """
+
+    workspace_id: str
+    api_key: str
+    api_root: str = CLOUD_API_ROOT
+
+    @property
+    def workspace_url(self) -> str | None:
+        return f"{self.api_root}/workspaces/{self.workspace_id}"
+
+    # Test connection and creds
+
+    def connect(self) -> None:
+        """Check that the workspace is reachable and raise an exception otherwise.
+
+        Note: It is not necessary to call this method before calling other operations. It
+              serves primarily as a simple check to ensure that the workspace is reachable
+              and credentials are correct.
+        """
+        _ = get_workspace(
+            api_root=self.api_root,
+            api_key=self.api_key,
+            workspace_id=self.workspace_id,
+        )
+        print(f"Successfully connected to workspace: {self.workspace_url}")
+
+    # Deploy and delete sources
+
+    def deploy_source(
+        self,
+        source: Source,
+    ) -> str:
+        """Deploy a source to the workspace.
+
+        Returns the newly deployed source ID.
+        """
+        source_configuration = source.get_config().copy()
+        source_configuration["sourceType"] = source.name.replace("source-", "")
+
+        deployed_source = create_source(
+            name=f"{source.name.replace('-', ' ').title()} (Deployed by PyAirbyte)",
+            api_root=self.api_root,
+            api_key=self.api_key,
+            workspace_id=self.workspace_id,
+            config=source_configuration,
+        )
+
+        # Set the deployment Ids on the source object
+        source._deployed_api_root = self.api_root  # noqa: SLF001  # Accessing nn-public API
+        source._deployed_workspace_id = self.workspace_id  # noqa: SLF001  # Accessing nn-public API
+        source._deployed_source_id = deployed_source.source_id  # noqa: SLF001  # Accessing nn-public API
+
+        return deployed_source.source_id
+
+    def delete_source(
+        self,
+        source: str | Source,
+    ) -> None:
+        """Delete a source from the workspace.
+
+        You can pass either the source ID `str` or a deployed `Source` object.
+        """
+        if not isinstance(source, (str, Source)):
+            raise ValueError(f"Invalid source type: {type(source)}")  # noqa: TRY004, TRY003
+
+        if isinstance(source, Source):
+            if not source._deployed_source_id:  # noqa: SLF001
+                raise ValueError("Source has not been deployed.")  # noqa: TRY003
+
+            source_id = source._deployed_source_id  # noqa: SLF001
+
+        elif isinstance(source, str):
+            source_id = source
+
+        delete_source(
```

Comment:
i am guessing the logic for making sure source is not part of any active connections exists in the api code.

Reply:
@bindipankhudi - Good question! I assumed so, but now I'm not 100% sure... 🤔